### PR TITLE
Fix rather a lot of typos

### PR DESCRIPTION
**Pull request by connortann**
Fixes a lot of spelling errors throughout the codebase.

Mostly internal fixes, but there is one public API change: renaming `shap.explainers.other.Coefficent` to `Coefficient`. For backwards compatibility, the original name is retained but deprecated.

### CODE REVIEW

File: shap/cext/_cext.cc [hunk 1]
```diff
@@ -209,7 +209,7 @@ static PyObject *_cext_dense_tree_shap(PyObject *self, PyObject *args)
     tfloat *base_offset = (tfloat*)PyArray_DATA(base_offset_array);
 
     // these are just a wrapper objects for all the pointers and numbers associated with
-    // the ensemble tree model and the datset we are explaing
+    // the ensemble tree model and the dataset we are explaing
```
Comment (thatlittleboy) [inline shap/cext/_cext.cc]:
```suggestion
    // the ensemble tree model and the dataset we are explaining
```

File: shap/cext/_cext.cc [hunk 2]
```diff
@@ -329,7 +329,7 @@ static PyObject *_cext_dense_tree_predict(PyObject *self, PyObject *args)
     tfloat *out_pred = (tfloat*)PyArray_DATA(out_pred_array);
 
     // these are just wrapper objects for all the pointers and numbers associated with
-    // the ensemble tree model and the datset we are explaing
+    // the ensemble tree model and the dataset we are explaing
```
Comment (thatlittleboy) [inline shap/cext/_cext.cc]:
```suggestion
    // the ensemble tree model and the dataset we are explaining
```

File: shap/cext/_cext.cc [hunk 3]
```diff
@@ -424,7 +424,7 @@ static PyObject *_cext_dense_tree_update_weights(PyObject *self, PyObject *args)
     bool *X_missing = (bool*)PyArray_DATA(X_missing_array);
 
     // these are just wrapper objects for all the pointers and numbers associated with
-    // the ensemble tree model and the datset we are explaing
+    // the ensemble tree model and the dataset we are explaing
```
Comment (thatlittleboy) [inline shap/cext/_cext.cc]:
```suggestion
    // the ensemble tree model and the dataset we are explaining
```

File: shap/cext/_cext.cc [hunk 4]
```diff
@@ -530,7 +530,7 @@ static PyObject *_cext_dense_tree_saabas(PyObject *self, PyObject *args)
     tfloat *out_pred = (tfloat*)PyArray_DATA(out_pred_array);
 
     // these are just wrapper objects for all the pointers and numbers associated with
-    // the ensemble tree model and the datset we are explaing
+    // the ensemble tree model and the dataset we are explaing
```
Comment (thatlittleboy) [inline shap/cext/_cext.cc]:
```suggestion
    // the ensemble tree model and the dataset we are explaining
```

File: shap/cext/_cext_gpu.cc [hunk 5]
```diff
@@ -150,7 +150,7 @@ static PyObject *_cext_dense_tree_shap(PyObject *self, PyObject *args)
     tfloat *base_offset = (tfloat*)PyArray_DATA(base_offset_array);
 
     // these are just a wrapper objects for all the pointers and numbers associated with
-    // the ensemble tree model and the datset we are explaing
+    // the ensemble tree model and the dataset we are explaing
```
Comment (thatlittleboy) [inline shap/cext/_cext_gpu.cc]:
```suggestion
    // the ensemble tree model and the dataset we are explaining
```

File: shap/cext/tree_shap.h [hunk 6]
```diff
@@ -1169,7 +1169,7 @@ inline void dense_independent(const TreeEnsemble& trees, const ExplanationDatase
     time_t start_time = time(NULL);
     tfloat last_print = 0;
     for (unsigned oind = 0; oind < trees.num_outputs; ++oind) {
-        // set the values int he reformated tree to the current output index
+        // set the values int he reformatted tree to the current output index
```
Comment (thatlittleboy) [inline shap/cext/tree_shap.h]:
```suggestion
        // set the values in the reformatted tree to the current output index
```

File: shap/explainers/other/__init__.py [hunk 7]
```diff
@@ -1,5 +1,13 @@
...
 from ._lime import LimeTabular
 from ._maple import Maple, TreeMaple
 from ._random import Random
 from ._treegain import TreeGain
+
+
+# Deprecated class alias with incorrect spelling
+def Coefficent(*args, **kwargs):  # noqa
+    warnings.warn("Coefficent has been renamed to Coefficient", DeprecationWarning)
+    return Coefficient(*args, **kwargs)
```
Comment (thatlittleboy) [inline shap/explainers/other/__init__.py]: Let's also set a version for when we will remove this deprecated class entirely, and mention it in the warning. How about `0.45.0`?

**Review by thatlittleboy — CHANGES_REQUESTED** [review]

File: shap/explainers/other/__init__.py (hunk 7 above)
Comment (connortann) [reply shap/explainers/other/__init__.py]: Good idea, done.

**Review by thatlittleboy — APPROVED** [review]
borat-great-success

**Comment by thatlittleboy** [comment]
@connortann Just curious, are you able to build the docs locally after this change?

I did `cd docs && make clean && make html`, but I'm getting the following error:

```
ERROR: [numpydoc] While processing docstring for 'shap.explainers.other.Coefficent'

Extension error (numpydoc.numpydoc):
Handler <function mangle_docstrings at 0x104a00c10> for event 'autodoc-process-docstring' threw an exception (exception: Expected a class or None, but got <function Coefficent at 0x11f0305e0>)
```

Not quite sure why the online RtD build is able to succeed..

**Comment by connortann** [comment]
I'm on mobile at the moment so I can't test myself unfortunately. That's surprising, as the old class shouldn't be referenced in the docs (as far as I can see) and I don't see any warnings in the RtD docs.

I wonder if there is some build artefact that is left over, which references the old variable name. Could you perhaps try with a clean clone in a new directory?

If you still get the issue, maybe one workaround could be to change the implementation of the deprecation from a function to a class, perhaps overriding `__new__` to return the new class?

**Comment by thatlittleboy** [comment]
> I wonder if there is some build artefact that is left over, which references the old variable name. Could you perhaps try with a clean clone in a new directory?

Yep a clean clone works. Maybe `make clean` doesn't clean all the artifacts.. >,<

**Comment by connortann** [comment]
That would be good to investigate - we could find out what is left behind and make sure it's added to `make clean`. I think by default Sphinx reuses intermediate files as a cache.

**Comment by thatlittleboy** [comment]
`make clean` only cleans the `_build` directory.

The `generated/` directory was left-over, I suspect that's the reason why. https://github.com/sphinx-doc/sphinx/issues/1999

**Pull request merged**